MIPs/mip-3.md

```diff
@@ -0,0 +1,184 @@
+---
+MIP: 3
+Title: Implement `wallet_swapAsset` to support Dapps to swap
+Status: Draft
+Stability: Experimental
+discussions-to: https://github.com/MetaMask/metamask-improvement-proposals/discussions/46
+Author(s): Tomás Santos (@tommasini)
+Type: Community
+Created: 2023-11-10
+---
+
+## Summary
+
+This proposal introduces a new JSON RPC method `wallet_swapAsset` that allows dApps to request MetaMask to perform a token swap operation. The method accepts parameters for the source and destination tokens and initiates a token swap operation.
```

Comment:
nit: `JSON RPC` should be connected by a dash. `JSON-RPC`

Reply:
Thanks!!! Replaced everywhere on this [commit](https://github.com/MetaMask/metamask-improvement-proposals/pull/25/commits/55c71d31a08464e00b9a3cfb055d09f17eca025d)